lib/pdn/pdn.c

```diff
@@ -310,22 +330,38 @@ static int cgact(uint8_t cid, bool activate)
 	return 0;
 }
 
-int pdn_activate(uint8_t cid, int *esm)
+int pdn_activate(uint8_t cid, int *esm, enum pdn_fam *fam)
 {
 	int err;
 	int timeout;
 
+	k_mutex_lock(&pdn_act_mutex, K_FOREVER);
+
+	pdn_act_notif.cid = cid;
+	k_sem_reset(&pdn_act_notif.sem_cgev);
+	k_sem_reset(&pdn_act_notif.sem_cnec);
+
 	err = cgact(cid, true);
+	if (!err && fam) {
+		k_sem_take(&pdn_act_notif.sem_cgev, K_FOREVER);
```

Comment:
We could have a Kconfigurable timeout for this, much like for the CNEC notification.
Would it make sense to use one Kconfig for timing out on both CNEC and CGEV notifications?

Reply:
The CGEV sem is only for the activation event, which we expect to always occur if the AT command succeeds. So I don't think a timeout is necessary there, whereas the CNEC timeout is needed because an ESM code is not always present.